File: Npc/11000296.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11000296: Jasper
/// </summary>
public class _11000296 : NpcScript {
    protected override int First() {
        return 30;
    }

    // Select 0:
    // $script:0831180407001177$
    // - What is it?
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (30, 0):
                // $script:0831180407001180$
                // - Oh! I's so happy to see another human being here!
                return -1;
        }
        
        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (30, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11000296.py
""" 11000296: Jasper """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0831180407001177$
        # - What is it?
        return None # TODO

    def __30(self, pick: int) -> int:
        # $script:0831180407001180$
        # - Oh! I's so happy to see another human being here!
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.CLOSE
        return Option.NONE
